### ISSUE TITLE
Bug: Weird text replacement with checkbox inside of editor

### ISSUE THREAD

**Issue by johannesspohr** [facebook/draft-js#2838]
#### Do you want to request a *feature* or report a *bug*?
Bug

#### What is the current behavior?

I have a block component which renders a checkbox in the editor, and when clicking the checkbox without focusing the editor first, the content from the previous selected element (although out of the editor) gets copied into the first block element of the editor.

![text-replacement](https://user-images.githubusercontent.com/1831098/109211660-c9e25400-77ae-11eb-83e9-3e72512d2978.gif)
(Editor is right, the "Notes" on the left is just a div)

It happens with checkboxes as well as radiobuttons (I also tried unstyled version). Buttons and other inputs are not causing the behavior. I also managed to reproduce it on the [medium-draft demo](http://bitwiser.in/medium-draft/), which offers checkboxes as well (I'm not using medium-draft though, just plain draft-editor 11).

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. You can use this jsfiddle to get started: https://jsfiddle.net/gmertk/e61z7nfa/.

https://jsfiddle.net/ueLyogb3/9/
Click inside the editor, click the "other Text", check the checkbox, Editor now says "Other Text"
![text-replacement2](https://user-images.githubusercontent.com/1831098/109213537-03b45a00-77b1-11eb-950f-45a7c124c552.gif)

#### What is the expected behavior?
The checkbox should be checked without the content being overwritten.

#### Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?

Tested with Chrome & Firefox on Windows, Draft 0.10 and 0.11 (latest)

**Comment by thaoms** [comment]
Did you notice that on Draft 0.11 you lose your focus on your text when toggling the checkbox and it's fine in 0.10?
That was actually the issue I was trying to report. But it does look that Draft js is dead.

**Comment by johannesspohr** [comment]
Yes, unfortunately no real activity here or on the code side :-/ (only dependabot keeps doing its job)
I think the only solution would be to replace the checkbox with a custom implementation and try to mimic the checkbox functionality as good as possible with aria rules etc. - not nice but probably would solve the issues

**Comment by thaoms** [comment]
I found a fix for myself, not sure if it will help you.

<input type='checkbox' defaultChecked={checked} onMouseDown={onChangeChecked} />

I made the checkbox not controlled because I'm maintaining the checked value in my block data anyway. And because it's mousedown instead of onchange I can easily preventDefault() and that fixed the focus thing.
